src/match/tasks.py

```diff
@@ -0,0 +1,75 @@
+import datetime
+import logging
+import os
+import sys
+
+from celery import Celery
+from celery.signals import after_setup_logger
+from django.utils import timezone
+from django_celery_beat.models import IntervalSchedule
+from django_celery_beat.models import PeriodicTask
+from match import match_status
+from match.controllers.create_match_controller import CreateMatchController
+from match.models import Match
+from person.models import Person
+
+# Get Pear algorithm
+current_dir = os.path.dirname(os.path.abspath(__file__))
+submodule_path = f"{current_dir}/../../pear-algorithm/src"
+sys.path.insert(0, submodule_path)
+from main import main
+
+app = Celery()
+
+# https://www.distributedpython.com/2018/08/28/celery-logging/
+logger = logging.getLogger(__name__)
+
+
+@after_setup_logger.connect
+def setup_loggers(logger, *args, **kwargs):
+    formatter = logging.Formatter(
+        "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+    )
+    sh = logging.StreamHandler()
+    sh.setFormatter(formatter)
+    logger.addHandler(sh)
+
+
+@app.task
+def matcher():
+    # Cancel previous matches that were not completed. TODO verify this is what we want to do
+    today = timezone.now()
+    last_week = today - datetime.timedelta(
+        days=7
+    )  # should be before pears are released but after matching
+    previous_matches = Match.objects.filter(created_date__range=[last_week, today])
+    print(f"Previous: {previous_matches}")
+    unfinished_matches = previous_matches.exclude(
+        status__in=[match_status.CANCELED, match_status.INACTIVE]
+    )
+    print(f"Unfinished: {unfinished_matches}")
+    unfinished_matches.update(status=match_status.CANCELED)
+
+    # Generate pears
+    users = Person.objects.all()
+    pears = main(users, logger)
+    match_creator = CreateMatchController(None, None, return_status=True)
+
+    # Create new matches
+    for pear in pears:
+        match_creator._data = {"ids": pear}
+        success, error_msg = match_creator.process()
+        if not success:
+            print(f"Match error between {pear}: {error_msg}")
+
+
+schedule, _ = IntervalSchedule.objects.get_or_create(
+    every=1,
```

Comment:
Is there a reason to do this every 1 minute? I know it's for testing purposes, but it just seems very frequent

Reply:
Stress test? lol, I feel like if I were deving and testing matching issues and I had to wait every 5-10 minutes I would be annoyed but happy to hear your thoughts @ReadeP @vianxnguyen

Reply:
Is it easy to start/stop when we're not using it for testing though?

Reply:
The Django beats service simply needs to be terminated and started up again. Although, this got me thinking that it would be nice if we can also change the interval on demand. I'll make a GitHub issue for this convenience.